docs/articles/forecast-windows.md

```diff
@@ -42,15 +42,32 @@ Forecast Window deadline, the TROLIE server returns `202 Accepted` to confirm
 that the `PATCH` contained a valid proposal and will be processed by the
 Clearinghouse.
 
+{: .nb }
+> **NOTE**: The Ratings Provider populates the `proposal-header.proposal-begins`
+> to indicate the intended Forecast Window. Accordingly, the TROLIE server
+> checks `proposal-header.proposal-begins` to determine if the proposal is late or not.
+
```

Comment:
So... devil's advocate here.  Without having the value, you can determine this implicitly simply by looking at the earliest period in the PATCH request body.  If you add the value, then the server must implement two validations:
1.  Enforce the invariant that all period starts are >= proposal-begins.
2.  What does the system do if there is a gap between "proposal-begins" and the first period in the PATCH?  Is it still valid?  What should the server do if "proposal-begins" is late, but none of the actual periods are?  

I would think that these discussions are unavoidable, and certainly represent work.  So, what's the tradeoff?  What do we gain from having it be explicitly called out in the header that is worth the work we've created?

Reply:
If I send you a forecast proposal just after a deadline has past but some of the forecast start on the hour appropriate to the current forecast window and others start in the prior hour--what was my intent?

If the header makes that explicit, there's no ambiguity.

Re: the validations, the first is a check we're doing already since `proposal-begins == forecast-window-first-hour` and the later is checked against each resource forecast to determine if it is late. For second point, the whole idea with this is that we need a way for the provider to be unambiguous about which Forecast Window they're targeting.

Reply:
Please note I renamed this to `begins` from `proposal-begins` because the way this is factored, that field shows up in all headers (request/response, RT/forecast). That was intentional--they all benefit from the explicit identification of the operational window--but the "proposal" part clearly doesn't make sense in all contexts.

Reply:
> If I send you a forecast proposal just after a deadline has past but some of the forecast start on the hour appropriate to the current forecast window and others start in the prior hour--what was my intent?
> 
> If the header makes that explicit, there's no ambiguity.
> 
> Re: the validations, the first is a check we're doing already since `proposal-begins == forecast-window-first-hour` and the later is checked against each resource forecast to determine if it is late. For second point, the whole idea with this is that we need a way for the provider to be unambiguous about which Forecast Window they're targeting.

I get it now.  Sold.